.github/workflows/resolve-image-tag.yaml

```
@@ -0,0 +1,48 @@
+name: Resolve image tag
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        description: "Path to rock directory with rockcraft.yaml"
+        required: true
+        type: string
+
+    outputs:
+      rock-reference:
+        description: "<rock-name>:<version> format"
+        value: ${{ jobs.resolve.outputs.rock-reference }}
+      full-image-tag:
+        description: "Fully qualified image tag"
+        value: ${{ jobs.resolve.outputs.full-image-tag }}
+
+jobs:
+  resolve:
+    name: Resolve tag ${{ inputs.rock-dir }}
+    runs-on: ubuntu-22.04
```

suggestion: Newer Ubuntu version
```suggestion
    runs-on: ubuntu-24.04
```